web/src/main/java/org/springframework/security/web/util/matcher/OrRequestMatcher.java

```diff
@@ -81,6 +82,23 @@ public MatchResult matcher(HttpServletRequest request) {
 		return MatchResult.notMatch();
 	}
 
+	@Override
+	public boolean equals(Object o) {
```

Comment:
Will you please put this in a preceding commit and then also add `hashCode` and `equals` for `AndRequestMatcher` in that commit?